Test/PublishPowerShellScript.Tests.ps1

```diff
@@ -0,0 +1,493 @@
+        ThenFailed -WithError '"Path\b.*\bis mandatory'
+        ThenScriptNotPublished -To $context.OutputDirectory
+    }
+}
+
+Describe 'PublishPowerShellScript.when path does not exist' {
+    AfterEach { Reset }
+    It 'should fail' {
+        Init
+        WhenPublishing -WithNonExistentPath -ErrorAction SilentlyContinue
+        ThenFailed -WithError 'does\ not\ exist'
+        ThenScriptNotPublished -To $context.OutputDirectory
+    }
+}
+
+Describe 'PublishPowerShellScript.when path is not a file' {
+    AfterEach { Reset }
+    It 'should fail' {
+        Init
+        WhenPublishing -WithInvalidPath -ErrorAction SilentlyContinue
+        ThenFailed -WithError 'should resolve to a file'
+        ThenScriptNotPublished -To $context.OutputDirectory
+    }
+}
+
+Describe 'PublishPowerShellScript.when script manifest is invalid' {
+    AfterEach { Reset }
+    It 'should fail' {
+        Init
+        WhenPublishing -ForManifestPath 'fubar' -ErrorAction SilentlyContinue
+        ThenFailed -WithError '"fubar"\ does\ not\ exist'
+        ThenScriptNotPublished -To $context.OutputDirectory
+    }
+}
+
+Describe 'PublishPowerShellScript.when given repository by location and by name' {
+    AfterEach { Reset }
+    It 'should use repository registered by location' {
+        Init
+        GivenRepository -Named 'RS7' -At 'RS7'
+        WhenPublishing -ToRepo 'RS8' -RepoAt $script:publishRoot
+        ThenSucceeded
+        ThenRepository 'RS8' -NotRegistered -NotExists
+        ThenRepository 'RS7' -NotRegistered -Exists
+        ThenScriptPublished -To 'RS7'
+    }
+}
+
+Describe 'PublishPowerShellScript.when given a credential ID' {
+    AfterEach { Reset }
+    It 'should pass' {
+        Init
+        $password = ConvertTo-SecureString 'MySecretPassword' -AsPlainText -Force
+        $credential = New-Object System.Management.Automation.PSCredential ('TestUser', $password)
+        $credentialID = [Guid]::NewGuid().ToString()
+        GivenCredential -Credential $credential -WithID $credentialID
+        Mock -CommandName 'Publish-Script' `
+             -ModuleName 'Whiskey' `
+             -Verifiable
+        WhenPublishing -WithCredentialID $credentialID
+        ThenSucceeded
+        Assert-VerifiableMock
+    }
+}
+
+Describe 'PublishPowerShellScript.when given a credential ID' {
+    AfterEach { Reset }
+    It 'should pass' {
+        Init
+        $password = ConvertTo-SecureString 'MySecretPassword' -AsPlainText -Force
+        $credential = New-Object System.Management.Automation.PSCredential ('TestUser', $password)
+        $credentialID = [Guid]::NewGuid().ToString()
+        GivenCredential -Credential $credential -WithID $credentialID
+        Mock -CommandName 'Publish-Script' `
+             -ModuleName 'Whiskey' `
+             -Verifiable
+        WhenPublishing -WithCredentialID $credentialID
+        ThenSucceeded
+        Assert-VerifiableMock

```

Comment:
Add an `Assert-MockCalled` that checks that the credential passed to Publish-Script is the one this test creates.